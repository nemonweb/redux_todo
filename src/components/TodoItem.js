import React, { Component } from 'react'
import PropTypes from 'prop-types'
export default class TodoItem extends Component {
  handleClick = () => {
    this.props.onClick && this.props.onClick(this.props.index)
  }

  render() {
    return (
      <li onClick={this.handleClick}>
        {this.props.text} {this.props.completed ? '+' : '-'}
      </li>
    )
  }
}

TodoItem.propTypes = {
  index: PropTypes.number,
  onClick: PropTypes.func,
  text: PropTypes.string,
  completed: PropTypes.bool
}
