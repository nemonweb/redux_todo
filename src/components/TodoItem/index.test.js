import React from 'react'
import TodoItem from './index'

describe('TodoItem', () => {
  it('should properly render self when completed false', () => {
    const wrapper = shallow(
        <TodoItem index={0} onClick={jest.fn()} text="xxx" completed={false} />
    );
    expect(wrapper).toMatchSnapshot();
  })

  it('should properly render self when completed true', () => {
    const wrapper = shallow(
        <TodoItem index={0} onClick={jest.fn()} text="xxx" completed={true} />
    );
    expect(wrapper).toMatchSnapshot();
  })

  it('should call onClick with prover attributes', () => {
    const onClickHandler = jest.fn()

    const wrapper = shallow(
      <TodoItem index={10} onClick={onClickHandler} text="xxx" completed={true} />
    );

    wrapper.props().onClick()
    expect(onClickHandler.mock.calls.length).toBe(1)
    expect(onClickHandler.mock.calls[0][0]).toBe(10)
  })
})
