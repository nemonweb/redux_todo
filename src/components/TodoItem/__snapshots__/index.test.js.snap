// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should properly render self when completed false 1`] = `
<li
  onClick={[Function]}
>
  xxx
   
  -
</li>
`;

exports[`TodoItem should properly render self when completed true 1`] = `
<li
  onClick={[Function]}
>
  xxx
   
  +
</li>
`;
